.cursor/rules/submission-contracts.mdc

```diff
@@ -0,0 +1,169 @@
+---
+description: 
+globs: 
+alwaysApply: true
+---
+---
+description: Guidelines for implementing submission contracts in WAVS components
+globs: src/contracts/**/*.sol
+---
+# Submission Contract Guidelines
+
+Guidelines for implementing submission contracts that handle WAVS component outputs.
+
+## Basic Structure
+
+1. Required Interfaces:
+```solidity
+// Required imports
+import {IWavsServiceManager} from "@wavs/interfaces/IWavsServiceManager.sol";
+import {IWavsServiceHandler} from "@wavs/interfaces/IWavsServiceHandler.sol";
+
+// Contract declaration
+contract YourSubmitContract is IWavsServiceHandler {
+    // Service manager instance
+    IWavsServiceManager private _serviceManager;
+
+    constructor(IWavsServiceManager serviceManager) {
+        _serviceManager = serviceManager;
+    }
+}
+```
+
+2. Required State Variables:
+```solidity
+// Mapping for tracking valid triggers
+mapping(TriggerId _triggerId => bool _isValid) internal _validTriggers;
+
+// Mapping for storing trigger data
+mapping(TriggerId _triggerId => bytes _data) internal _datas;
+
+// Mapping for storing operator signatures
+mapping(TriggerId _triggerId => bytes _signature) internal _signatures;
+```
+
+## Core Implementation
+
+1. Handle Signed Data Function:
+```solidity
+function handleSignedData(bytes calldata _data, bytes calldata _signature) external {
+    // 1. Validate operator signature
+    _serviceManager.validate(_data, _signature);
+
+    // 2. Decode data into DataWithId struct
+    DataWithId memory dataWithId = abi.decode(_data, (DataWithId));
+
+    // 3. Store results
+    _signatures[dataWithId.triggerId] = _signature;
+    _datas[dataWithId.triggerId] = dataWithId.data;
+    _validTriggers[dataWithId.triggerId] = true;
```

Comment:
Think this can just be a comment... you check the signature, get the data, and do something with it.